src/doc/tutorials/PythonExpressions.rst

```diff
@@ -0,0 +1,280 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Python expression we just created can also be used with the **CLI**.
+
+We will start by saving the script we just created.
+
+1. Click *Save script* and save the script with the name ``MyExpression.py``.
+
+The following script will open ``curv2d.silo`` and create a Pseudocolor plot of the expression.
+
+.. code-block:: python
+
+  OpenDatabase("/usr/gapps/visit/data/curv2d.silo")
+
+  DefinePythonExpression("MyExpression", ['d', 'p'], file="MyExpression.py")
+
+  AddPlot("Pseudocolor", "MyExpression")
+  DrawPlots()
+
+Python Expression Example 2
+---------------------------
+
+This example operates on 2D meshes and takes the distance around the edges of each cell and multiplies it by the value of the cell.
+It demonstrates accessing the coordinates and topology of the mesh as well as a variable.
+
+Here is the example script.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 18-61
+
+Let us start off by creating a Pseudocolor plot from the expression.
+
+1. Copy the script into the *Python expression script* editor.
+2. Click the *Apply* button.
+3. Go to *Variables->MyExpression* to change the variable to the expression.
+
+.. figure:: images/PythonExpressions-Plot2.png
+
+   The Pseudocolor plot of MyExpression
+
+The ``__init__`` and ``modify_contract`` methods are the same as the previous example, so we will only look at the ``derive_variable`` method.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 28-29
+
+The following lines get the ``vtkDataArray`` for the cell values and the number of cells.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 30-33
+
+The following lines set the output ``vtkDataArray`` to be an array of floats with 1 component and ncells values.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 34-37
+
+Now we loop over the cells, setting the output value for each cell.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 38
+
+The following lines get the current cell and the number of edges in the cell.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 39-42
+
+Now we loop over the edges, calculating the sum of the lengths of the edges.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 43-45
+
+We calculate the length of the edge from the 3D coordinates of the end points of the edge, which we add to the sum.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 46-54
+
+Once we have summed the lengths of the edges we multiply the sum by the cell value and set it in the result.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 55-58
+
+Once we have finished processing all the cells, we return the ``vtkDataArray``.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_2.vpe
+   :lines: 59
```

Comment:
Another observation here...instead of interleaving a line of prose documenting what is happening in reStructurecText and literal including a few lines from the actual `.vpe` file and going back and forth between these many times (8 times above I think), why not just do the documentation (as comments...there are some comments there that more or less duplicate what is written here in reStructuredText) and literalinclude the whole file, perahpts with a `:start-at:` directive that bypasses the boilerplate VisIt comment header.